src/values.rs

```diff
@@ -745,8 +747,8 @@ impl ValueBuilder for CoreTypeConcrete {
             CoreTypeConcrete::Uint32(_) => false,
             CoreTypeConcrete::Uint64(_) => false,
             CoreTypeConcrete::Uint128(_) => false,
-            CoreTypeConcrete::Uint128MulGuarantee(_) => todo!(),
-            CoreTypeConcrete::NonZero(_) => todo!(),
+            CoreTypeConcrete::Uint128MulGuarantee(_) => false,
+            CoreTypeConcrete::NonZero(_) => false,
```

Comment:
What about `NonZero<felt252>`? We should recurse for types that have inner data mimicking its layout.
I already implemented the fix in my AOT PR and changing it here would create more merge conflicts there.